benchmark/checkperf.cmake

```diff
@@ -7,7 +7,7 @@
 
 # Clone the repository if it's not there
 find_package(Git QUIET)
-if (SIMDJSON_IS_UNDER_GIT AND Git_FOUND AND (GIT_VERSION_STRING VERSION_GREATER  "2.1.4") AND (NOT CMAKE_GENERATOR MATCHES Ninja) ) # We use "-C" which requires a recent git
+if (SIMDJSON_IS_UNDER_GIT AND SIMDJSON_GIT AND Git_FOUND AND (GIT_VERSION_STRING VERSION_GREATER  "2.1.4") AND (NOT CMAKE_GENERATOR MATCHES Ninja) ) # We use "-C" which requires a recent git
```

Comment:
Not for this PR, but are `SIMDJSON_IS_UNDER_GIT AND SIMDJSON_GIT AND Git_FOUND` always used in tandem? Like, can we make SIMDJSON_GIT be the combination of these three? Like, if the user says SIMDJSON_GIT=ON, we error out if git is not there or it's not under a git repository ... and if it's not set, we set it to SIMDJSON_GIT=ON iff we're under a .git repository and git is found. Then everybody doesn't have to add all these checks, just check SIMDJSON_GIT.

Just a theoretical question since I know you've been messing around in the various places we mess with git.